src/graph/entities/qg_node.c

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2018-2019 Redis Labs Ltd. and Contributors
+ *
+ * This file is available under the Redis Labs Source Available License Agreement
+ */
+
+#include "qg_node.h"
+#include "../graph.h"
+#include "../../util/arr.h"
+#include <assert.h>
+
+QGNode* QGNode_New(const char *label, const char *alias) {
+    QGNode *n = rm_malloc(sizeof(QGNode));
+    n->alias = NULL;
+    n->label = NULL;
+    n->labelID = GRAPH_NO_LABEL;
+    n->incoming_edges = array_new(QGEdge*, 0);
+    n->outgoing_edges = array_new(QGEdge*, 0);
+    // n->id = -1;
```

Comment:
Maybe we should set ID to uninitializeID,

Reply:
I think I will make ID an argument into this function; it's decided by the QueryGraph logic.